lib/mix/lib/mix/scm/git.ex

```diff
@@ -125,14 +125,18 @@ defmodule Mix.SCM.Git do
     sparse_toggle(opts)
     update_origin(opts[:git])
 
+    rev = get_lock_rev(opts[:lock], opts) || get_opts_rev(opts)
```

Comment:
Moved this up because `git fetch --depth=N` needs the target ref to be fetched. When we would fallback to the default branch, we simply omit `rev` (see the [`if` expression in line 135](https://github.com/elixir-lang/elixir/pull/13128/files#diff-e79c1de922ff4acc89a122e0567970c9eebb6fdf49fdae26322f7b6a214dffa6R135)).